File: cpp/StructuralPatterns/DecoratorPattern/Problem/BasicPizza.cpp
#ifndef BASIC_PIZZA_CPP
#define BASIC_PIZZA_CPP

#include "Pizza.cpp"

class BasicPizza : public Pizza {
public:
    std::string getDescription() override {
        return "Basic Pizza";
    }

    double getCost() override {
        return 5.00; // base cost
    }
};

#endif 

File: cpp/StructuralPatterns/DecoratorPattern/Solution/BasicPizza.java
package org.Prathamesh.StructuralPatterns.DecoratorPattern.Solution;

public class BasicPizza implements Pizza {
    @Override
    public String getDescription() {
        return "Basic Pizza";

    }

    @Override
    public double getCost() {
        return 5.00; //base cost
    }
}
